apollo-router/src/services/subgraph_service.rs

```diff
@@ -945,6 +1052,85 @@ mod tests {
         server.await.unwrap();
     }
 
+    // starts a local server emulating a subgraph returning bad response format
+    async fn emulate_subgraph_ok_status_invalid_response(socket_addr: SocketAddr) {
```

Comment:
nit: instead of taking a (static) socket address as argument, that will end up conflicting with something else, this should open a listen socket on a random port and return the address

Reply:
Fixed all tests [6aec2e0](https://github.com/apollographql/router/pull/3328/commits/6aec2e00d41f4d24f2aa274783473a4b03a23b64)